src/tdastro/base_models.py

```diff
@@ -63,39 +62,81 @@ def add_parameter(self, name, value=None, required=False, **kwargs):
         cannot be found.
         Raise a ``ValueError`` if the parameter is required, but set to None.
         """
-        if hasattr(self, name) and getattr(self, name) is not None:
-            raise KeyError(f"Duplicate parameter set: {name}")
+        # Check for parameter has been added and if so, find the index.
+        try:
+            ind = next(ind for ind, entry in enumerate(self.setters) if entry[0] == name)
```

Comment:
Wouldn't dictionary do the `self.setters`' job better? Dictionary guarantees to iterate in the insertion order.

Reply:
I didn't know dictionaries were guaranteed to maintain insertion order. Let me merge this PR and send you a followup to change that over.